packages/vm/tests/api/types.spec.ts

```diff
@@ -0,0 +1,43 @@
+import tape from 'tape'
+import Common from '@ethereumjs/common'
+import {
+  AccessListEIP2930Transaction,
+  AccessListEIP2930TxData,
+  Transaction,
+  TxData,
+} from '@ethereumjs/tx'
+import { Block, BlockData } from '@ethereumjs/block'
+
+tape('[Types]', function (t) {
+  t.test('should ensure that the actual objects can be safely used as their data types', (st) => {
+    type RequiredExceptOptionals<TypeT, OptionalFieldsT extends keyof TypeT> = Required<
+      Omit<TypeT, OptionalFieldsT>
+    > &
+      Pick<TypeT, OptionalFieldsT>
+
+    const common = new Common({ chain: 'mainnet', hardfork: 'berlin' })
+
+    // Block
+    const block: Required<BlockData> = Block.fromBlockData({}, { common })
+    st.ok(block, 'block')
+
+    // Transactions
+    type OptionalTxFields = 'to' | 'r' | 's' | 'v'
+
+    // Legacy tx
+    const legacyTx: RequiredExceptOptionals<TxData, OptionalTxFields> = Transaction.fromTxData(
+      {},
+      { common }
+    )
+    st.ok(legacyTx, 'legacy tx')
+
+    // Access List tx
+    const accessListTx: RequiredExceptOptionals<
+      AccessListEIP2930TxData,
+      OptionalTxFields
+    > = AccessListEIP2930Transaction.fromTxData({}, { common })
+    st.ok(accessListTx, 'accessList tx')
+
+    st.end()
+  })
+})
```

Comment:
Super-impressive new type of tests (I am barely able to decipher this though TBH 😬 , have some intuition what it does though 😋 )

Reply:
Update: ah, maybe you *can* add some 2-3 comment here upon the `type` definition and the appliance to give some hints how this actually works for fellow devs.

Reply:
yeah this was from @alcuadrado who whipped up this magic in the discord 🙂 i will add some comments as dev notes.

Reply:
added in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1185/commits/b16fd4daa910c76bf19b10866d71b1f4e33192df, let me know if the comment explains sufficiently

Reply:
Yeah, that's good for now.